store/dynamodb/dynamodb_test.go

```diff
@@ -0,0 +1,210 @@
+	)
+	assert.NoError(t, err)
+	assert.NotNil(t, kv)
+
+	if ddb, ok := kv.(*DynamoDB); !ok {
+		t.Fatal("Error registering and initializing DynamoDB")
+		ddb.createTable()
+	}
+
+}
+
+func TestSetup(t *testing.T) {
+	ddb := newDynamoDBStore(t)
+	// ensure this is idempotent
+	err := ddb.createTable()
+	assert.Nil(t, err)
+}
+
+func TestDynamoDBStore(t *testing.T) {
+	ddbStore := newDynamoDBStore(t)
+	backupStore := newDynamoDBStore(t)
+	testutils.RunTestCommon(t, ddbStore)
+	testutils.RunTestAtomic(t, ddbStore)
+	testutils.RunTestTTL(t, ddbStore, backupStore)
+}
+
+func TestDynamoDBStoreLock(t *testing.T) {
+	ddbStore := newDynamoDBStore(t)
+	backupStore := newDynamoDBStore(t)
+	testutils.RunTestLock(t, ddbStore)
+	testutils.RunTestLockTTL(t, ddbStore, backupStore)
+}
+
+func TestDynamoDBStoreUnsupported(t *testing.T) {
+	ddbStore := newDynamoDBStore(t)
+
+	_, err := ddbStore.WatchTree("test", nil, nil)
+	assert.Equal(t, store.ErrCallNotSupported, err)
+
+	_, err = ddbStore.Watch("test", nil, nil)
+	assert.Equal(t, store.ErrCallNotSupported, err)
+}
+
+func TestBatchWrite(t *testing.T) {
+
+	dynamodbSvc := newDynamoDB()
+
+	mock := &mockedBatchWrite{DynamoDBAPI: dynamodbSvc}
+	mock.BatchWriteResp = &dynamodb.BatchWriteItemOutput{
+		UnprocessedItems: map[string][]*dynamodb.WriteRequest{
+			"test-1-valkeyrie": []*dynamodb.WriteRequest{
+				&dynamodb.WriteRequest{
+					DeleteRequest: &dynamodb.DeleteRequest{
+						Key: map[string]*dynamodb.AttributeValue{
+							"id": &dynamodb.AttributeValue{
+								S: aws.String("abc123"),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	mock.Count = 1
+
+	kv := &DynamoDB{
+		dynamoSvc: mock,
+		tableName: "test-1-valkeyrie",
+	}
+
+	prefix := "testDeleteTree"
+
+	firstKey := "testDeleteTree/first"
+	firstValue := []byte("first")
+
+	secondKey := "testDeleteTree/second"
+	secondValue := []byte("second")
+
+	// Put the first key
+	err := kv.Put(firstKey, firstValue, nil)
+	assert.NoError(t, err)
+
+	// Put the second key
+	err = kv.Put(secondKey, secondValue, nil)
+	assert.NoError(t, err)
+
+	err = kv.DeleteTree(prefix)
+	assert.NoError(t, err)
+}
+
+type mockedBatchWrite struct {
+	dynamodbiface.DynamoDBAPI
+	BatchWriteResp *dynamodb.BatchWriteItemOutput
+	Count          int
+}
+
+func TestDecodeItem(t *testing.T) {
+	data := map[string]*dynamodb.AttributeValue{
+		partitionKey: &dynamodb.AttributeValue{
+			S: aws.String("abc123"),
+		},
+		revisionAttribute: &dynamodb.AttributeValue{
+			N: aws.String("10"),
+		},
+		encodedValueAttribute: &dynamodb.AttributeValue{
+			S: aws.String("YWJjMTIzCg=="),
+		},
+	}
+
+	kv, err := decodeItem(data)
+	assert.NoError(t, err)
+	assert.Equal(t, &store.KVPair{Key: "abc123", Value: []uint8{0x61, 0x62, 0x63, 0x31, 0x32, 0x33, 0xa}, LastIndex: 0xa}, kv)
+
+	data[encodedValueAttribute] = &dynamodb.AttributeValue{S: aws.String("not base64")}
+	kv, err = decodeItem(data)
+	assert.Error(t, err)
+	assert.Nil(t, kv)
+}
+
+func TestPutWithTTL(t *testing.T) {
```

Comment:
The `TestPutWithTTL` is already in `utils` so we could remove it in DynamoDB tests?